### ISSUE TITLE
Snowflake Tag Association - Object Identifier Mapping Incorrect

### ISSUE THREAD

**Issue by joshtree41** [Snowflake-Labs/terraform-provider-snowflake#1558]
**Provider Version**

0.56.3

**Terraform Version**

1.3.9

**Describe the bug**

When using the `snowflake_tag_association` resource to associated a _tag_ to _column(s)_.

The `object_identifier` block is not properly capturing the object.

Example:

```hcl
resource "snowflake_tag" "pii-tag" {
  provider       = snowflake.sys
  name           = "PII_TAG"
  database       = module.security.db.name
  schema         = snowflake_schema.pii-schema.name
  allowed_values = toset(["SSN", "DOB", "NAME", "EMAIL", "ADDRESS", "PHONE", "PRN", "TIN", "EIN", "ID", "CARD NUMBER", "OTHER"])
}

resource "snowflake_tag_association" "table_association" {
  tag_id      = snowflake_tag.pii-tag.id
  object_type = "COLUMN"
  tag_value   = "VALUE"
  object_identifier {
    database = "DB"
    schema   = "SCHEMA"
    name     = "TABLE.COL"
  }
}
```

```
snowflake_tag_association.table_association: Creating...
╷
│ Error: error associating tag to object: ["DB"."SCHEMA"."TABLE.COL"] with command: [ALTER TABLE "SEC"."TAGS"."TABLE" MODIFY COLUMN COL SET TAG "SEC"."TAGS"."PII_TAG" = 'PRN'], tag_id [SEC|TAGS|PII_TAG]
│
│   with snowflake_tag_association.table_association,
│   on security.tf line 75, in resource "snowflake_tag_association" "table_association":
│   75: resource "snowflake_tag_association" "table_association" {
│
╵
make: *** [apply] Error 1
```

The object looks like `["DB"."SCHEMA"."TABLE.COL"]` but the rendered Sql defaults to the location of of the tag (in this case `"SEC"."TAGS"`).

```sql
ALTER TABLE "SEC"."TAGS"."TABLE" MODIFY COLUMN COL SET TAG "SEC"."TAGS"."PII_TAG" = 'PRN';
```

This issues makes it impossible to apply tags to objects that aren't colocated in the same schema.

**Expected behavior**

The above resource would render a SQL statement like 

```sql
ALTER TABLE "DB"."SCHEMA"."TABLE" MODIFY COLUMN COL SET TAG "SEC"."TAGS"."PII_TAG" = 'PRN'
```

**Code samples and commands**

Please add code examples and commands that were run to cause the problem.

**Additional context**

**Comment by joshtree41** [comment]
Can I get some love over here? I think this would help some folks out. Column level governance in terraform is one PR away.

**Comment by sfc-gh-ngaberel** [comment]
@sfc-gh-kumaurya @joshtree41 is this fixed?

**Comment by sfc-gh-kumaurya** [comment]
@sfc-gh-ngaberel Yes, this will be available in the next release v0.64.0.

**Comment by joshtree41** [comment]
Oh yeahhhhh, thank you @sfc-gh-kumaurya! I'll test it out after I see the new version pop up
